.pre-commit-config.yaml

```diff
@@ -96,3 +49,24 @@ repos:
     rev: v0.8.0.4
     hooks:
       - id: shellcheck # TODO remove this when all shell scripts have been removed from otx
+
+  # markdown linting
+  - repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.33.0
+    hooks:
+      - id: markdownlint
+        args: [--config=.markdownlint.yaml]
+
+  # Ruff
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: "v0.0.258"
+    hooks:
+      - id: ruff
+        exclude: "tests|otx/recipes"
+        args: [--fix]
+
+  # # Git conflict marker
+  # - repo: https://github.com/jumanjihouse/pre-commit-hooks
```

Comment:
If you're asking about the --fix option, it would be used for the developer's env while making a commit. and the git conflict marker linting hook will be enabled in the separated PR because it would be fixed too much files for the whitespace related things.